### PR TITLE
feat(csharp): add github action for release

### PR DESCRIPTION
**Pull request by morganleroi**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1717](https://algolia.atlassian.net/browse/DI-1717)

### Changes included:
- Add Github Action Workflow to release a new Nugget package.

[DI-1717]: https://algolia.atlassian.net/browse/DI-1717?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/algoliasearch-client-csharp/.github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+on:
+  push:
+    branches:
+      - next
+
+jobs:
+  release:
+    name: Publish
+    runs-on: ubuntu-latest
```
Comment (millotp) [inline clients/algoliasearch-client-csharp/.github/workflows/release.yml:11]: we should pin the ubuntu version

File: clients/algoliasearch-client-csharp/.github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    if: "startsWith(github.event.head_commit.message, 'chore: release')"
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
Comment (millotp) [inline clients/algoliasearch-client-csharp/.github/workflows/release.yml:16]: you don't want `fetch-depth: 0`, you can remove it and leave the default of 1 to only fetch the latest commit

File: clients/algoliasearch-client-csharp/.github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
```
Comment (millotp) [inline clients/algoliasearch-client-csharp/.github/workflows/release.yml:19]: should you pin the version for this ?

**Review by millotp — COMMENTED** [review]
nice

File: clients/algoliasearch-client-csharp/.github/workflows/release.yml
Comment (millotp) [inline clients/algoliasearch-client-csharp/.github/workflows/release.yml:1]: you need to add this file to `generation.config.js`, it's being override by the CI

File: clients/algoliasearch-client-csharp/.github/workflows/release.yml (hunk 3 above)
Comment (morganleroi) [reply clients/algoliasearch-client-csharp/.github/workflows/release.yml:17]: By default, it uses the `global.json`file. I use the global json property to enforce the documentation of what we are doing here.

File: clients/algoliasearch-client-csharp/.github/workflows/release.yml
Comment (morganleroi) [reply clients/algoliasearch-client-csharp/.github/workflows/release.yml:1]: ah yes done.

**Review by millotp — APPROVED** [review]
perfect !

**Pull request merged**